theme/icons.go

```diff
@@ -569,13 +569,16 @@ func (t *builtinTheme) Icon(n fyne.ThemeIconName) fyne.Resource {
 	return icons[n]
 }
 
-// ThemedResource is a resource wrapper that will return a version of the resource with the main color changed
-// for the currently selected theme.
+// ThemedResource is a resource wrapper used to represent a version of a resource.
+// It changes colors based on the active theme.
+// This is currently achieved using XML attributes.
+// This is designed for use with Icons of 24px x 24px.
 type ThemedResource struct {
 	source fyne.Resource
 }
 
-// NewThemedResource creates a resource that adapts to the current theme setting.
+// NewThemedResource creates a resource that adapts to the current theme primary color.
+// Receives any struct implementing the Resource interface of Content() and Name()
```

Comment:
Is this needed? the parameter is `Resource` so it is clear that the parameter must implement `Resource`.

Reply:
I was trying for this (from GoLand) in VSCode. I clearly slipped and hit my head 😄 
![image](https://user-images.githubusercontent.com/2605791/147729408-f767f39d-756c-45ac-b3aa-afe8a9d5859c.png)

Reply:
I agree with @andydotxyz. Seems a bit unnecessary.